public/app/features/search/page/components/FolderSection.tsx

```diff
@@ -72,6 +76,16 @@ export const FolderSection = ({
   }, [sectionExpanded, tags]);
 
   const onSectionExpand = () => {
+    const lastExpandedFolder = window.localStorage.getItem(SEARCH_EXPANDED_FOLDER_STORAGE_KEY);
+
+    if (sectionExpanded) {
+      if (lastExpandedFolder === uid) {
+        window.localStorage.removeItem(SEARCH_EXPANDED_FOLDER_STORAGE_KEY);
+      }
+    } else {
+      window.localStorage.setItem(SEARCH_EXPANDED_FOLDER_STORAGE_KEY, uid);
+    }
+
     setSectionExpanded(!sectionExpanded);
```

Comment:
Nit: I find it a bit harder to follow code when it uses a prop that's about to change, like here where if the section WAS expanded and is about to no longer be, we remove the item from localStorage.
Could we create a variable before the if `const newSectionExpanded = !sectionExpanded` and just use that in the if and to set it on line 89?
Also the `lastExpandedFolder` variable can just be created inside the first `if` since it's only used there.